### PR TITLE
Move to `hatchling` for build + `pyproject.toml` for package def

### LINKED ISSUE #1641: Use `hatchling` to build the package
> Hmm, git revision does not display in the about dialog of the GUI (seen while testing the docker image) - I'll investigate.

`_baked_revision.py` is present in the wheel, but not in the docker image. I think the proper fix is to use `hatchling` to build the package, like we are already doing for `LiberTEM-{live,blobfinder,...}`.

### PR DESCRIPTION
**Pull request by matbryan52**
Closes #1641. WIP to see how angry CI is at me for pulling the rug out from under it.

- [x] Moves project definition into `pyproject.toml` and use `hatchling` as backend
- [x] Remove `setup.cfg` by moving config options to other files as appropriate
- [x] Move the `build_client` and `copy_client` commands into a dedicated script with `click` for CLI
- [x] Remove `setup.py`
- [x] Adjust `README.rst` to bake in the citations, to keep `pyroma` happy and have correct rendering on GitHub and citations on `PyPi`.
- [x] ~~Use https://github.com/ofek/hatch-vcs as plugin to generate `__version__` and `revision` from tags.~~
  - [ ] ~~I don't like that the normal version number now has the git revision in it, I need to look more into this plugin to see how to configure it.~~
- [x] Port `hatch_build.py` from `LiberTEM-live`
- [ ] ~~Make `pyroma` happy again in CI due to lack of `setup.cfg`~~
- [x] Disable `pyroma` due to missing build dependencies in CI (only?)
- [ ] ~~Update the release scripts to handle setting the version and revision from git tag~~

## Contributor Checklist:

* [x] I have added or updated my entry in [the creators.json file](https://github.com/LiberTEM/LiberTEM/blob/master/packaging/creators.json)
* [x] I have added [a changelog entry](https://github.com/LiberTEM/LiberTEM/tree/master/docs/source/changelog) for my contribution
* [x] I have added/updated documentation for all user-facing changes
* [ ] I have added/updated test cases
* [ ] I have included the [rebuilt production build of the client](https://libertem.github.io/LiberTEM/contributing.html?#building-the-client) (only if changes were made to the GUI)

## Reviewer Checklist:

* [x] `/azp run libertem.libertem-data` passed
* [x] No import of GPL code from MIT code

### CODE REVIEW

**Comment by sk1p** [comment]
> * [ ]  I don't like that the normal version number now has the git revision in it, I need to look more into this plugin to see how to configure it.

You can also check how this is done for LiberTEM-live, which uses a custom `hatch_build.py` for baking the revision, and `tool.hatch.version` for reading the version from `__version__.py`  (basically keeps the old way of versioning, just with new packaging around it)

**Comment by matbryan52** [comment]
> > * [ ]   I don't like that the normal version number now has the git revision in it, I need to look more into this plugin to see how to configure it.
> 
> You can also check how this is done for LiberTEM-live, which uses a custom `hatch_build.py` for baking the revision, and `tool.hatch.version` for reading the version from `__version__.py` (basically keeps the old way of versioning, just with new packaging around it)

Great, my fault for not looking too closely at the way you set it up over there. I've followed the same pattern here! Thanks for the tip.

**Review by sk1p — CHANGES_REQUESTED** [review]
:+1: looks good overall. One major thing: the wheel and sdist (built with `python -m build --sdist --wheel`) don't seem to include `_baked_revision.py`. Not sure why yet...

File: tox.ini [hunk 1]
```diff
@@ -3,9 +3,9 @@ envlist = py{39,310,311,312}{,-data}{,-nogpu}, notebooks
 
 [testenv]
 commands=
-    pytest --durations=10 --cov=libertem --cov-report=term --cov-report=html --cov-report=xml --cov-config=setup.cfg --junitxml=junit.xml {posargs:tests/}
+    pytest --durations=10 --cov=libertem --cov-report=term --cov-report=html --cov-report=xml --cov-config=pyproject.toml --junitxml=junit.xml {posargs:tests/}
     # win_tweaks.py depends on modules that are only available on Windows
-    pytest --doctest-modules --ignore=src/libertem/common/win_tweaks.py --ignore=src/libertem/analysis/fullmatch.py src/libertem/
+    pytest --doctest-modules --ignore=src/libertem/common/win_tweaks.py --ignore=src/libertem/_baked_revision.py --ignore=src/libertem/analysis/fullmatch.py src/libertem/
```
[the remark this answers is not in the document]
Comment (matbryan52) [reply tox.ini:8]: I was getting CI failures for lack of coverage information on `_baked_revision.py`. I will try removing it again now that the _baked_revision is in the sdist.

[the remark this answers is not in the document]
Comment (matbryan52) [reply tox.ini]: Still failing when I don't ignore `_baked_revision.py` in doctest-modules : https://github.com/LiberTEM/LiberTEM/actions/runs/9795236694/job/27046926599?pr=1667

[the remark this answers is not in the document]
Comment (sk1p) [reply tox.ini]: Yeah - it's weird because IIRC this doesn't happen in LiberTEM-live...

[the remark this answers is not in the document]
Comment (sk1p) [reply tox.ini]: Might still be a difference between what's installed in the test environment vs. what's in the source tree?

[the remark this answers is not in the document]
Comment (sk1p) [reply tox.ini]: Yeah, sdist includes the baked revision, the wheel doesn't?

[the remark this answers is not in the document]
Comment (sk1p) [reply tox.ini]:
```diff
diff --git a/pyproject.toml b/pyproject.toml
index c715e6fa..bd50fd93 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -111,6 +111,11 @@ Homepage = "https://libertem.github.io/LiberTEM/"
 [tool.hatch.version]
 path = "src/libertem/__version__.py"
 
+[tool.hatch.build.targets.wheel]
+artifacts = [
+    "/src/libertem/_baked_revision.py",
+]
+
 [tool.hatch.build.targets.sdist]
 artifacts = [
     "/src/libertem/_baked_revision.py",
@@ -181,4 +186,4 @@ exclude_lines = [
     "if 0:",
     "if False:",
     "if __name__ == .__main__.:",
```
?

[the remark this answers is not in the document]
Comment (matbryan52) [reply tox.ini]: OK, I will find the right magic words :D

**Review by sk1p — APPROVED** [review]
LGTM!

**Pull request merged**